src/rust/engine/src/nodes.rs

```diff
@@ -1205,13 +1205,9 @@ impl Node for NodeKey {
 
   fn run(self, context: Context) -> NodeFuture<NodeResult> {
     let span_id = generate_random_64bit_string();
-    let node_workunit_params = if context.session.should_record_zipkin_spans() {
-      let node_name = format!("{}", self);
-      let start_time = std::time::SystemTime::now();
-      Some((node_name, start_time, span_id.clone()))
-    } else {
-      None
-    };
+    let node_name = format!("{}", self);
```

Comment:
Super nitpick, but even if you don't need the scope for the if/else reason, I still kind of like:
   
```
    let node_workunit_params = {
        let node_name = format!("{}", self);
        let start_time = std::time::SystemTime::now();
        (node_name.clone(), start_time, span_id.clone())
    };
```
as that really clearly indicates that `node_name` and `start_time` won't be needed further down this function, so it reduces the cognitive overhead on the reader of tracking a couple more variables.